addon/model/util/model-tree-walker.ts

```diff
@@ -26,14 +26,26 @@ enum TraverseType {
 
 export type ModelNodeFilter = (node: ModelNode) => FilterResult;
 
+export function toFilterSkipFalse(func: (node: ModelNode) => boolean): ModelNodeFilter {
+  return function (node: ModelNode) {
+    return func(node) ? FilterResult.FILTER_ACCEPT : FilterResult.FILTER_SKIP;
+  };
+}
+
+export function toFilterRejectFalse(func: (node: ModelNode) => boolean): ModelNodeFilter {
+  return function (node: ModelNode) {
+    return func(node) ? FilterResult.FILTER_ACCEPT : FilterResult.FILTER_REJECT;
+  };
+}
+
 export interface ModelTreeWalkerConfig {
   filter?: ModelNodeFilter;
   range: ModelRange;
   descend?: boolean;
   visitParentUpwards?: boolean;
 }
 
-export default class ModelTreeWalker implements Iterable<ModelNode> {
+export default class ModelTreeWalker<T extends ModelNode = ModelNode> implements Iterable<T> {
```

Comment:
ah I can use this in my offset to range mapper :dancers:

Reply:
its still a cast though so be careful
but yea very useful when your filter only outputs one type

Reply:
its why the nodefinder used to have both a filter and a predicate
(filter was for types, predicate for other stuff)
but I found it too confusing and usually redundant